glyphs-reader/src/glyphdata_bundled.rs

```diff
@@ -75,7 +75,12 @@ fn name_offset(idx: usize) -> usize {
     if idx == NAME_OFFSETS.len() {
         return NAMES.len();
     }
-    NAME_OFFSETS.get(idx).unwrap()
+    NAME_OFFSETS.get(idx).unwrap_or_else(|| {
+        panic!(
+            "Asked for name {idx} but we only have {} names",
+            NAMES.len()
+        )
+    })
```

Comment:
I'd just make this an assert at the top of the fn, then unwrap down here.

Reply:
I'm failing to grasp why that's an improvement?

Reply:
I am inferring that it is an invariant of this function that `idx` is in range. If you want to make this invariant explicit, the way I would generally expect that to be done is with an assert at the top of the fn; then the rest of the function can be read knowing the invariants, and it can focus more on the actual logic.

As a secondary point, this is just not what `unwrap_or_else` is really for; the idea is that you're using the closure to compute a fallback value, not to abort.


So `unwrap` is fine, `expect` is fine, but if we're trying to get fancy with a custom panic message, that's what asserts are for.